internal/ocm/client.go

```diff
@@ -545,18 +545,36 @@ func NewOpenShiftVersionXY(v string) string {
 }
 
 // NewOpenShiftVersionXYZ parses the given version and converts it to CS readable version
-func NewOpenShiftVersionXYZ(v string) string {
+// CS readable version "openshift-v<X.Y.Z>" or "openshift-v<X.Y.Z>-channel_group"
+func NewOpenShiftVersionXYZ(v, cg string) string {
 	var csVersion string
 
 	if len(v) > 0 {
-		parts := strings.Split(v, ".")
+		// Separate version from prerelease (e.g., "4.19.0" from "0.nightly-2025-01-01")
+		versionPart := v
+		prereleasePart := ""
+		if hyphenIndex := strings.Index(v, "-"); hyphenIndex != -1 {
+			versionPart = v[:hyphenIndex]
+			prereleasePart = v[hyphenIndex:] // includes the hyphen
+		}
+
+		parts := strings.Split(versionPart, ".")
 		if len(parts) == 1 {
 			parts = append(parts, "0")
 		}
-		// FIXME This assumes X.Y is 4.19. Eventually
-		//       we may need a switch statement here.
-		parts = append(parts[:2], OpenShift419Patch)
-		csVersion = api.OpenShiftVersionPrefix + strings.Join(parts, ".")
+
+		// If no patch version provided (X.Y format), append default patch version
+		// Otherwise preserve the provided patch version (X.Y.Z format)
+		if len(parts) == 2 {
+			parts = append(parts, OpenShift419Patch)
+		}
+
+		csVersion = api.OpenShiftVersionPrefix + strings.Join(parts, ".") + prereleasePart
+
+		// Only append channel group if it's not empty and not "stable"
+		if len(cg) > 0 && cg != "stable" {
+			csVersion = csVersion + "-" + cg
+		}
```

Comment:
Could you add a comment somewhere in here showing the expected version formats for the various channels we're now supporting?  I gather from the code it's going to be something like:
```
   stable: openshift-vX.Y.Z
candidate: openshift-vX.Y.Z[-prereleasePart]-candidate
  nightly: openshift-vX.Y.Z[-prereleasePart]-nightly
```

But I'd like more detail about how `prereleasePart` is formatted.

In the example you gave in a comment above it looks like "nightly" could appear twice?
e.g. `openshift-v4.19.0-0.nightly-2025-01-01-nightly`?